homeassistant/components/mqtt/light/schema_json.py

```diff
@@ -452,6 +452,14 @@ async def async_turn_off(self, **kwargs):
         if ATTR_TRANSITION in kwargs:
             message["transition"] = kwargs[ATTR_TRANSITION]
 
+        if ATTR_FLASH in kwargs:
+            flash = kwargs.get(ATTR_FLASH)
+
+            if flash == FLASH_LONG:
+                message["flash"] = self._flash_times[CONF_FLASH_TIME_LONG]
+            elif flash == FLASH_SHORT:
+                message["flash"] = self._flash_times[CONF_FLASH_TIME_SHORT]
+
```

Comment:
Move all of this to a new helper which is called both from `async_turn_off` and `async_turn_on`.

Reply:
Done

Reply:
Great, can you name the method `_set_flash_and_transition` instead and handle both?